tests/tools/test_memory_monitor.py

```diff
@@ -0,0 +1,192 @@
+
+    assert isinstance(memory_data, dict)
+    assert MemoryType.RSS in memory_data
+    assert MemoryType.SYSTEM in memory_data
+    if return_max_value:
+        assert all(map(lambda v: isinstance(v, float), memory_data.values()))
+    else:
+        assert all(map(lambda v: isinstance(v, tuple) and len(v) == 2 and len(v[0]) == len(v[1]), memory_data.values()))
+
+    saved_files = tuple(tmpdir.glob("*"))
+    assert len(saved_files) == 8
+    assert sum(map(lambda fn: int(str(fn).endswith(".txt")), saved_files)) == 4
+    assert sum(map(lambda fn: int(str(fn).endswith(".png")), saved_files)) == 4
+
+
+@pytest.mark.skipif(is_windows(), reason="Check on linux only")
+def test_empty_logs(tmpdir):
+    memory_monitor = MemoryMonitor().start()
+    memory_monitor.stop()
+    memory_monitor._memory_values_queue = queue.Queue()  # make sure no logs are recorded
+    time_values, memory_values = memory_monitor.get_data()
+    assert len(time_values) == len(memory_values) == 0
+    memory_monitor.save_memory_logs(time_values, memory_values, Path(tmpdir))
+
+
+@pytest.mark.skipif(ISOLATION_RUN_ENV_VAR not in os.environ, reason="Should be run via isolation proxy")
+def test_memory_values_isolated():
+    baseline_memory = float(os.environ[BASELINE_MEMORY_VAR]) if BASELINE_MEMORY_VAR in os.environ else None
+
+    memory_monitor = MemoryMonitor(memory_type=MemoryType.RSS, memory_unit=MemoryUnit.B).start()
+    bytes_to_allocate = 1 if baseline_memory is None else BYTES_TO_ALLOCATE_LARGE
+    allocate(bytes_to_allocate, sleep_before_deallocation=True, sleep_after_deallocation=True)
+    memory_monitor.stop()
+    _, memory_values = memory_monitor.get_data()
+
+    if baseline_memory is None:
+        print("\nMax memory:", max(memory_values))
+    else:
+        memory_values = list(map(lambda it: it - baseline_memory, memory_values))
+        rel = 1e-2
+        assert max(memory_values) == pytest.approx(BYTES_TO_ALLOCATE_LARGE, rel=rel)
+        assert abs(memory_values[0]) < BYTES_TO_ALLOCATE_LARGE * rel
+        assert abs(memory_values[-1]) < BYTES_TO_ALLOCATE_LARGE * rel
+
+
+@pytest.mark.skipif(is_windows(), reason="Check on linux only")
+def test_memory_values():
+    # The first run of the test collects the memory that is allocated by default
+    _, stdout, _ = run_pytest_case_function_in_separate_process(test_memory_values_isolated)
+    max_mem_line = next(filter(lambda s: "Max memory:" in s, stdout.split("\n")))
+    baseline_memory = max_mem_line.split(" ")[-1]
+
+    # The second run of the test checks that the added amount of memory is correctly represented by the memory monitor
+    os.environ[BASELINE_MEMORY_VAR] = baseline_memory
+    run_pytest_case_function_in_separate_process(test_memory_values_isolated)
+    del os.environ[BASELINE_MEMORY_VAR]
+
+
+@pytest.mark.skipif(ISOLATION_RUN_ENV_VAR not in os.environ, reason="Should be run via isolation proxy")
```

Comment:
Are the tests with `ISOLATION_RUN_ENV_VAR` launched in CI?

Reply:
Yes, this env. variable is set by `run_pytest_case_function_in_separate_process()`